File: src/types/media.ts
export interface MediaItem {
  path: string;
  filename: string;
  size_bytes: number;
  duration_secs?: number;
  format?: string;
  resolution?: string;
  category: MediaCategory;
}

export type MediaCategory = "movies" | "shows" | "music" | "other";

export interface EnrichedMediaItem {
  path: string;
  title: string;
  original_filename: string;
  year?: number;
  overview?: string;
  poster_path?: string;
  local_poster_path?: string;
  runtime?: number;
  genres: string[];
  tmdb_id?: number;
  media_type: TMDBMediaType;
  season?: number;
  episode?: number;
  tag_id?: string;
}

export type TMDBMediaType = "movie" | "tv" | "unknown";

export interface TMDBSearchResult {
  id: number;
  title: string;
  overview: string;
  release_date?: string;
  poster_path?: string;
  vote_average?: number;
  media_type: string;
}

export interface ParsedFilename {
  title: string;
  year?: number;
  season?: number;
  episode?: number;
  media_type: "Movie" | "TVShow" | "Unknown";
}

File: src-tauri/src/models/media.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct MediaItem {
    pub path: String,
    pub filename: String,
    pub size_bytes: u64,
    pub duration_secs: Option<u32>,
    pub format: Option<String>,
    pub resolution: Option<String>,
    pub category: MediaCategory,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "lowercase")]
pub enum MediaCategory {
    Movies,
    Shows,
    Music,
    Other,
}

impl Default for MediaCategory {
    fn default() -> Self {
        Self::Other
    }
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct EnrichedMediaItem {
    pub path: String,
    pub title: String,
    pub original_filename: String,
    pub year: Option<u32>,
    pub overview: Option<String>,
    pub poster_path: Option<String>,
    pub local_poster_path: Option<String>,
    pub runtime: Option<u32>,
    pub genres: Vec<String>,
    pub tmdb_id: Option<u64>,
    pub media_type: TMDBMediaType,
    pub season: Option<u32>,
    pub episode: Option<u32>,
    pub tag_id: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "lowercase")]
pub enum TMDBMediaType {
    Movie,
    Tv,
    Unknown,
}

impl Default for TMDBMediaType {
    fn default() -> Self {
        Self::Unknown
    }
}
